src/types/UiaTextRangeBase.cpp

```diff
@@ -233,14 +159,49 @@ const IdType UiaTextRangeBase::GetId() const noexcept
     return _id;
 }
 
-const Endpoint UiaTextRangeBase::GetStart() const noexcept
+const COORD UiaTextRangeBase::GetEndpoint(TextPatternRangeEndpoint endpoint) const noexcept
 {
-    return _start;
+    switch (endpoint)
+    {
+    case TextPatternRangeEndpoint::TextPatternRangeEndpoint_End:
+        return _end;
+    case TextPatternRangeEndpoint::TextPatternRangeEndpoint_Start:
+    default:
+        return _start;
+    }
 }
 
-const Endpoint UiaTextRangeBase::GetEnd() const noexcept
+// Routine Description:
+// - sets the target endpoint to the given COORD value
+// - if the target endpoint crosses the other endpoint, become a degenerate range
+// Arguments:
+// - endpoint - the target endpoint (start or end)
+// - val - the value that it will be set to
+// Return Value:
+// - true if range is degenerate, false otherwise.
+bool UiaTextRangeBase::SetEndpoint(TextPatternRangeEndpoint endpoint, const COORD val) noexcept
 {
-    return _end;
+    const auto bufferSize = _pData->GetTextBuffer().GetSize();
+    switch (endpoint)
+    {
+    case TextPatternRangeEndpoint::TextPatternRangeEndpoint_End:
+        _end = val;
+        if (bufferSize.CompareInBounds(_end, _start, true) < 0)

```

Comment:
If... end is left of start then set start to end so it is degenerate at the new end point? Right?

Maybe leave that comment so I don't have to dissect that mentally again.